### PR TITLE
Add OASv2 and OASv3 languages

### PR DESCRIPTION
**Pull request by whuji**
## Description

Add OASv2 and OASv3 disambiguation as heuristics for JSON and YAML (see https://github.com/github/linguist/issues/6115)

## Checklist:

- [x] **I am adding a new language.**
  - [x] The extension of the new language is used in hundreds of repositories on GitHub.com.
  - [x] I have included a real-world usage sample for all extensions added in this PR
  - [x] I have included a syntax highlighting grammar: (I have reused YAML and JSON grammars)
  - [x] I have updated the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -4381,6 +4382,64 @@ Nunjucks:
   tm_scope: text.html.nunjucks
   ace_mode: nunjucks
   language_id: 461856962
+OASv2:
+  type: programming
```
Comment (lildude) [inline lib/linguist/languages.yml:4386]: These are not programming languages, they're specifications written in JSON or YAML, which are both data types, so this should be `data` too. If a repo owner wants the name to appear in the language stats, they can implement an override.

```suggestion
  type: data
```

File: lib/linguist/languages.yml [hunk 2]
```diff
@@ -4381,6 +4382,64 @@ Nunjucks:
...
+  color: "#85ea2d"
+  tm_scope: none
+  ace_mode: text
+  language_id: 848295328
+OASv2-json:
+  type: programming
```
Comment (lildude) [inline lib/linguist/languages.yml:4392]:
```suggestion
  type: data
```

File: lib/linguist/languages.yml [hunk 3]
```diff
@@ -4381,6 +4382,64 @@ Nunjucks:
...
+  extensions:
+  - ".json"
+  group: OASv2
+  tm_scope: source.json
+  ace_mode: json
+  codemirror_mode: javascript
+  codemirror_mime_type: application/json
+  language_id: 834374816
+OASv2-yaml:
+  type: programming
```
Comment (lildude) [inline lib/linguist/languages.yml:4403]:
```suggestion
  type: data
```

File: lib/linguist/languages.yml [hunk 4]
```diff
@@ -4381,6 +4382,64 @@ Nunjucks:
...
+  - ".yaml"
+  - ".yml"
+  group: OASv2
+  tm_scope: source.yaml
+  ace_mode: yaml
+  codemirror_mode: yaml
+  codemirror_mime_type: text/x-yaml
+  language_id: 105187618
+OASv3:
+  type: programming
```
Comment (lildude) [inline lib/linguist/languages.yml:4415]:
```suggestion
  type: data
```

File: lib/linguist/languages.yml [hunk 5]
```diff
@@ -4381,6 +4382,64 @@ Nunjucks:
...
+  color: "#85ea2d"
+  tm_scope: none
+  ace_mode: text
+  language_id: 557959099
+OASv3-json:
+  type: programming
```
Comment (lildude) [inline lib/linguist/languages.yml:4421]:
```suggestion
  type: data
```

File: lib/linguist/languages.yml [hunk 6]
```diff
@@ -4381,6 +4382,64 @@ Nunjucks:
...
+  extensions:
+  - ".json"
+  group: OASv3
+  tm_scope: source.json
+  ace_mode: json
+  codemirror_mode: javascript
+  codemirror_mime_type: application/json
+  language_id: 980062566
+OASv3-yaml:
+  type: programming
```
Comment (lildude) [inline lib/linguist/languages.yml:4432]:
```suggestion
  type: data
```

**Review by lildude — CHANGES_REQUESTED** [review]
See inline comments. 

The `ghes-3.4.*` samples are also massive. Please replace these with much smaller samples (ProTip: it the diff collapses them, they're too big 😉).

**Comment by whuji** [comment]
I have implemented all requested changes.
I know that OAS is data but I didn't know it was possible to configure GitHub to show that in languages card, that is why I set it to programming. In that regards, I didn't find how to override this setting, can you please guide me?

**Comment by lildude** [comment]
> In that regards, I didn't find how to override this setting, can you please guide me?

https://github.com/github/linguist/blob/master/docs/overrides.md#detectable

**Review by lildude — DISMISSED** [review]

**Review by lildude — CHANGES_REQUESTED** [review]
I've just noticed you've not added tests specifically for your new heuristics for matching the OAS files. Please add these.

**Comment by whuji** [comment]
@lildude Yes, just added them + rebase from master for f-f merging.

Thanks !

**Review by lildude — APPROVED** [review]

**Pull request merged**